internal/usecase/push_subscription_usecase.go

```
@@ -0,0 +1,116 @@
+		if err != nil {
+			// Log error but continue to send to other subscriptions
+			// If the error is due to a gone subscription, delete it from DB
+			// For simplicity, we'll just log for now.
+			// In a real application, you'd want to handle specific webpush errors
+			// e.g., webpush.ErrSubscriptionExpired, webpush.ErrSubscriptionGone
+			// and delete the subscription from the database.
+			// For now, we'll just log the error.
+			// fmt.Printf("Failed to send notification to %s: %v\n", sub.Endpoint, err)
+			// If the subscription is no longer valid, delete it
+			if err.Error() == "webpush: subscription is no longer valid" || err.Error() == "webpush: subscription is gone" {
+				u.subscriptionRepo.Delete(ctx, sub.Endpoint)
```

golangci-lint (1.64.8)

Error return value of `u.subscriptionRepo.Delete` is not checked

(errcheck)